types/src/proof/definition.rs

```diff
@@ -177,15 +177,10 @@ impl SparseMerkleProof {
         )
     }
 
-    /// If `element_hash` is present, verifies an element whose key is `element_key` and value is
-    /// authenticated by `element_hash` exists in the Sparse Merkle Tree using the provided proof.
-    /// Otherwise verifies the proof is a valid non-inclusion proof that shows this key doesn't
-    /// exist in the tree.
-    pub fn verify_by_hash(
+    pub fn verify_by_root_hash(
```

Comment:
nit: comment why this exists compared to the full fledged API. ("this is used internally just to verify the integrity of the authentication structure, doesn't defend against the case where one provides values that doesn't match the proof)

Reply:
done